archive_upload/handlers/dsmc_handlers.py

```diff
@@ -0,0 +1,754 @@
+            self.abort(msg)
+            return
+
+        dsmc_log_dir = "{}/dsmc_{}".format(dsmc_log_root_dir, runfolder_archive)
+
+        if not os.path.exists(dsmc_log_dir):
+            os.makedirs(dsmc_log_dir)
+
+        # Fetch the description of the last uploaded version of this archive
+        descr = helper.get_pdc_descr(path_to_archive, dsmc_log_dir)
+
+        # Get the local and remote filelist, and then get the list of files
+        # that are missing on remote side, or differs in byte size.
+        # NB. Uploaded list contains folders as well, but when we check local
+        # content we only look at the files, and ignore the folders.
+        uploaded_files = helper.get_pdc_filelist(path_to_archive, descr, dsmc_log_dir)
+        local_files = helper.get_local_filelist(path_to_archive)
+        reupload_files = helper.get_files_to_reupload(local_files, uploaded_files)
+
+        # Upload the missing files with the same description previously used.
+        if reupload_files:
+            job_id = helper.reupload(reupload_files, descr, dsmc_log_dir, self.runner_service)
+            log.debug("Reupload job_id {}".format(job_id))
+
+            status_end_point = "{0}://{1}{2}".format(
+            self.request.protocol,
+            self.request.host,
+            self.reverse_url("status", job_id))
+
+            response_data = {
+                "job_id": job_id,
+                "service_version": version,
+                "link": status_end_point,
+                "state": State.STARTED,
+                "dsmc_log_dir": dsmc_log_dir}
+
+            self.set_status(202, reason="started reuploading")
+        else:
+            log.debug("Nothing to do - everything already uploaded.")
+
+            response_data = {
+            "service_version": version,
+            "state": State.DONE,
+            "dsmc_log_dir": dsmc_log_dir}
+
+            self.set_status(200, reason="nothing to reupload")
+
+        self.write_object(response_data)
+
+class UploadHandler(BaseDsmcHandler):
+    """
+    Handler for uploading an archive to PDC.
+    """
+    def post(self, runfolder_archive):
+        """
+        Tells `dsmc` to upload `runfolder_archive` to PDC, with a uniquely generated description label.
+        Job is run in the background to be polled by the status endpoint.
+
+        :param runfolder_archive: the name of the archive that we want to upload
+        :return: HTTP 202 if the upload as started successfully, with a `job_id` to be used for later status polling,
+                 HTTP 500 if an unexpected error was encountered
+        """
+
+        monitored_dir = self.config["path_to_archive_root"]
+
+        if not self._validate_runfolder_exists(runfolder_archive, monitored_dir):
+            msg = "Error when validating runfolder. {} is not found under {}".format(runfolder_archive, monitored_dir)
+            self.abort(msg)
+            return
```

Comment:
Is the return necessary here?

Reply:
The returns were necessary to finish the requests. But I'll try to do it in the `abort()` method instead. Think it should work at least.